core/lib/dal/src/pruning_dal.rs

```diff
@@ -0,0 +1,345 @@
+use anyhow::anyhow;
+use sqlx::error::BoxDynError;
+use zksync_db_connection::{connection::Connection, instrument::InstrumentExt};
+use zksync_types::{L1BatchNumber, MiniblockNumber};
+
+use crate::Core;
+
+#[derive(Debug)]
+pub struct PruningDal<'a, 'c> {
+    pub(crate) storage: &'a mut Connection<'c, Core>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct PruningInfo {
+    pub last_soft_pruned_l1_batch: Option<L1BatchNumber>,
+    pub last_soft_pruned_miniblock: Option<MiniblockNumber>,
+    pub last_hard_pruned_l1_batch: Option<L1BatchNumber>,
+    pub last_hard_pruned_miniblock: Option<MiniblockNumber>,
+}
+
+#[derive(Debug, sqlx::Type)]
+#[sqlx(type_name = "prune_type")]
+pub enum PruneType {
```

Comment:
can we please have some comments on what Soft and Hard is in this context

Reply:
I've added a readme